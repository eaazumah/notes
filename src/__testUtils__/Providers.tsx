import React from 'react';
import {Provider} from 'react-redux';
import {ThemeProvider} from 'styled-components/native';
import useTheme from '../@shared/hooks/useTheme';
import {store} from '../@shared/redux/store';

interface IReduxProvider {
  children?: React.ReactNode;
}

export const TestReduxProvider = ({children}: IReduxProvider) => {
  return <Provider store={store}>{children}</Provider>;
};

interface ITestThemeProvider {
  children: React.ReactNode;
}

export const TestThemeProvider = ({children}: ITestThemeProvider) => {
  const {theme} = useTheme();
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

interface ITestProviders {
  children: React.ReactNode;
}

const TestProvider = ({children}: ITestProviders) => {
  return (
    <TestReduxProvider>
      <TestThemeProvider>{children}</TestThemeProvider>
    </TestReduxProvider>
  );
};

export default TestProvider;
