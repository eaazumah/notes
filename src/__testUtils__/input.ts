import faker from 'faker';
import {nanoid} from 'nanoid';

const items = Array(20).fill(1);

const getMockNotes = () => {
  return items.map(() => ({
    id: nanoid(),
    createdAt: Date.now(),
    title: faker.lorem.words(),
    text: faker.lorem.sentences(),
    favorite: faker.datatype.boolean(),
  }));
};

export default getMockNotes;
