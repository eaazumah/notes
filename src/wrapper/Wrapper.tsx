import React, {ReactNode} from 'react';
import {StatusBar} from 'react-native';
import {ThemeProvider} from 'styled-components/native';
import useTheme from '../@shared/hooks/useTheme';

interface Props {
  [x: string]: ReactNode;
}

const Wrapper = (props: Props) => {
  const {theme} = useTheme();

  return (
    <ThemeProvider theme={theme}>
      <StatusBar
        translucent
        barStyle={theme.barStyle}
        backgroundColor="transparent"
      />
      {props.children}
    </ThemeProvider>
  );
};

export default Wrapper;
