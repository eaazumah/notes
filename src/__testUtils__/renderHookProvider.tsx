import {renderHook, RenderHookOptions} from '@testing-library/react-hooks';
import React from 'react';
import {TestReduxProvider} from './Providers';

const renderHookReduxWrap = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  options?: Omit<RenderHookOptions<TProps>, 'wrapper'>,
) => {
  return renderHook(callback, {
    ...options,
    wrapper: ({children}) => <TestReduxProvider>{children}</TestReduxProvider>,
  });
};

export default renderHookReduxWrap;
