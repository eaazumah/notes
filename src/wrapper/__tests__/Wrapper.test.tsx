import {render} from '@testing-library/react-native';
import React from 'react';
import {TestReduxProvider} from '../../__testUtils__/Providers';
import Wrapper from '../Wrapper';

describe('Wrapper', () => {
  test('Wrapper renders correctly', () => {
    const wrapper = render(
      <TestReduxProvider>
        <Wrapper />
      </TestReduxProvider>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
