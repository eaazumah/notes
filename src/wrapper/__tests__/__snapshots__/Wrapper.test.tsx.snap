// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper Wrapper renders correctly 1`] = `null`;
